### PR TITLE
Don't override path in NavLink component. Fixes #6613

### LINKED ISSUE #6613: withRouter wrapped component cannot access params when used in NavLink
I've searched the related issues and I didn't find this exact bug. Apologies if I missed it.

The issue is that when I try to use a component wrapped with `withRouter` inside a `NavLink`, I don't get the correct `match` prop. Weirdly, this is only an issue with `NavLink`. Everything works correctly within a regular `Link`. 

## Version

React Router Dom 4.3.1

## Test Case

https://gist.github.com/zfletch/28bbaa330756607d40e4c2acf7d980b2

(Uncomment the `NavLink` code and click on the `person 2` to see it break).

## Steps to reproduce

1. Create a component using `withRouter`
2. Use `this.props.match.params` in that component
3. Place that component in a `<NavLink>` tag

## Expected Behavior

It should work like in any other component (including `<Link>`). The `match` and `params` should be present.

## Actual Behavior

`match` is `null` in some cases and `{}` in other cases. This causes incorrect behavior and errors when one of these components is used inside `<NavLink>`.

### PR DESCRIPTION
**Pull request by StringEpsilon**
See comments [here.](https://github.com/ReactTraining/react-router/issues/6613#issuecomment-472109397)

I added some unit tests in NavLink-test and also removed the location override as a precaution for other interactions.

### CODE REVIEW

**Comment by StringEpsilon** [comment]
@timdorr Added tests. Let me know of any feedback you might have.

**Comment by StringEpsilon** [comment]
BTW, you might want to check for any documentation or examples where you show or suggest using <Route/> as a wrapper / matching-utensil like in NavLink.

**Comment by StringEpsilon** [comment]
Rebased and added a similar refactoring as #6685 too. I can move that to it's own PR if you want to.

File: packages/react-router-dom/modules/NavLink.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import React from "react";
-import { Route, matchPath } from "react-router";
+import {
+  __RouterContext as RouterContext,
+  Route,
```
Comment (MeiKatz) [inline packages/react-router-dom/modules/NavLink.js]: `<Route />` can be removed as it is obsolete.

Comment (timdorr) [reply packages/react-router-dom/modules/NavLink.js]: It's in the PropTypes.

Comment (StringEpsilon) [reply packages/react-router-dom/modules/NavLink.js]: Oops

Comment (StringEpsilon) [reply packages/react-router-dom/modules/NavLink.js]: @timdorr Should it really be there? In both cases it's just PropTypes.bool. And it's rather unlikely to change.

**Comment by timdorr** [comment]
Can you update the size snapshot?

**Comment by StringEpsilon** [comment]
@timdorr  Rebased onto master and corrected the snapshots while doing so.

**Comment by timdorr** [comment]
Great! Thanks!

**Pull request merged**